### DISCUSSION TITLE
弹出层中新建表单使用switch的getFieldValue动态展示item问题

### DISCUSSION THREAD

**Discussion by hongcyu** [ant-design/ant-design#30183] [Q&A] [answered]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/danchucengzhongdexinjianbiaodan-antd4151-forked-xoyd6)

### Steps to reproduce
请看复现链接的第57-64行
        <Form.Item name="switch" label='switch'>
          <Switch />
        </Form.Item>
        {
          form.getFieldValue('switch') && 
          <Form.Item name="input" label='input'>
          <Input/>
          </Form.Item>
      }

### What is expected?
form.item中有个switch，当switch为true时才展示下面的input，为false时不展示input

### What is actually happening?
发现form.getFieldValue('switch') 在为undefined，但提交表单时却有值。想询问form.getFieldValue不是实时的吗？
有什么办法可以让switch为true时展示input，为false时不展示input呢

| Environment | Info |
|---|---|
| antd | 4.12.1 |
| React | 16.14.0 |
| System | macOS |
| Browser | chrome 88.0.4324.182 |

**Comment by jameslahm** [comment, marked as answer]
When changing `switch` value, `CollectionCreateForm` will not rerender. You could add `onChange` callback to determine whether show `input`. [demo here](https://codesandbox.io/s/danchucengzhongdexinjianbiaodan-antd4151-forked-lxjfx?file=/index.js)

**Reply by hongcyu** [reply]
Thanks.This solution is right！

**Comment by xzy-1115** [comment]
有同样的需求，但是我的表单控件是有config配置传入一个通用的formmodal生成，请问要如何解决
